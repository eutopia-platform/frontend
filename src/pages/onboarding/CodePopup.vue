<template>
  <div class="code-popup">
    <Popup>
      <div class="code-wrap">
        <Header type='secondary'>
          Find a secret in your email
        </Header>
        <Paragraph>
          We have send you a 6 digit confirmation code to {{info.email}}. It
          will expire shortly, so please enter your code soon.
        </Paragraph>
        <div class="input-wrap">
          <PinInput @submit="onSubmit"></PinInput>
        </div>
        <Paragraph>
          Keep this tab open to enter your code. If you didn't receive an email,
          check your spam folder.
        </Paragraph>
        <div class="icon-wrap" :src="img">
          <Icon :src="img" size="5rem"></Icon>
        </div>
      </div>
    </Popup>
  </div>
</template>

<script>
import Popup from '../../components/molecular/Popup.vue'
import Header from '../../components/atomic/Header.vue'
import Paragraph from '../../components/atomic/Paragraph.vue'
import PinInput from '../../components/atomic/PinInput.vue'
import Icon from '../../components/atomic/Icon.vue'

export default {
  name: "CodePopup",
  components: {
    Popup,
    Header,
    Paragraph,
    PinInput,
    Icon
  },
  props: {
    info: Object
  },
  methods: {
    onSubmit: function() {
      this.$emit('submit')
    }
  },
  computed: {
    img: () => require('../../../data/img/onboarding/inbox.svg')
  }
}
</script>

<style lang="scss" scoped>
.code-wrap {
  width: 40ch;
  padding-left: 3rem;
  padding-right: 3rem;
  padding-bottom: 3rem;

  .input-wrap {
    margin-top: 2rem;
    margin-bottom: 2rem;
    text-align: center;
  }

  .icon-wrap {
    margin-top: 2rem;
    text-align: center;
  }
}
</style>
