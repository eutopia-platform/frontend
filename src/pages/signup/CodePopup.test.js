import { mount } from '@vue/test-utils'
import CodePopup from './CodePopup'

describe('A CodePopup', () => {
  it('renders same thing as always', () => {
    const popup = mount(CodePopup, {
      stubs: ['Shade', 'Paragraph', 'PinInput', 'Icon'],
    })
    expect(popup).toMatchSnapshot()
  })
})
