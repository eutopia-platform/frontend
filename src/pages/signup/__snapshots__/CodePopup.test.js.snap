// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A CodePopup renders same thing as always 1`] = `
<div class="popup">
  <shade-stub></shade-stub>
  <div class="content">
    <div class="code-wrap">
      <header secondary="">Find a secret in your email</header>
      <paragraph-stub>
        We have sent you a 6 digit confirmation code to . It
        will expire shortly, so please enter your code soon.
      </paragraph-stub>
      <div class="input-wrap">
        <pininput-stub value=""></pininput-stub>
      </div>
      <paragraph-stub>
        Keep this tab open to enter your code. If you didn't receive an email,
        check your spam folder.
      </paragraph-stub>
      <div src="[object Object]" class="icon-wrap">
        <icon-stub src="[object Object]" size="5rem"></icon-stub>
      </div>
    </div>
  </div>
</div>
`;
